### ISSUE TITLE
import os; os.<tab> will trigger internal python language server error.

### ISSUE THREAD

**Issue by hongyi-zhao** [spyder-ide/spyder#11534]
Hi,

The latest git version of spyder, when do the following:

```
import os
os.<tab>
```

Will trigger the internal python language server error.

Issue reporter gives the following:

```
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/pyls_jsonrpc/endpoint.py", line 113, in consume
    self._handle_request(message['id'], message['method'], message.get('params'))
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/pyls_jsonrpc/endpoint.py", line 182, in _handle_request
    handler_result = handler(params)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/pyls_jsonrpc/dispatchers.py", line 23, in handler
    return method(**(params or {}))
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/pyls/python_ls.py", line 321, in m_text_document__completion
    return self.completions(textDocument['uri'], position)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/pyls/python_ls.py", line 239, in completions
    completions = self._hook('pyls_completions', doc_uri, position=position)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/pyls/python_ls.py", line 155, in _hook
    return hook_handlers(config=self.config, workspace=workspace, document=doc, **kwargs)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/pluggy/manager.py", line 337, in traced_hookexec
    return outcome.get_result()
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/pluggy/callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/pluggy/callers.py", line 52, in from_call
    result = func()
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/pluggy/manager.py", line 335, in <lambda>
    outcome = _Result.from_call(lambda: oldcall(hook, hook_impls, kwargs))
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/pluggy/manager.py", line 84, in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/pluggy/callers.py", line 208, in _multicall
    return outcome.get_result()
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/pluggy/callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/pyls/plugins/jedi_completion.py", line 49, in pyls_completions
    definitions = document.jedi_script(position).completions()
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/api/__init__.py", line 216, in completions
    return completion.completions()
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/api/completion.py", line 99, in completions
    completion_names = self._get_context_completions()
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/api/completion.py", line 207, in _get_context_completions
    completion_names += self._trailer_completions(dot.get_previous_leaf())
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/api/completion.py", line 258, in _trailer_completions
    completion_names += filter.values()
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/filters.py", line 228, in values
    return [n for filter in self._filters for n in filter.values()]
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/filters.py", line 228, in <listcomp>
    return [n for filter in self._filters for n in filter.values()]
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/filters.py", line 87, in values
    return self._convert_names(
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/filters.py", line 84, in _convert_names
    return [self.name_class(self.context, name) for name in names]
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/filters.py", line 84, in <listcomp>
    return [self.name_class(self.context, name) for name in names]
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/filters.py", line 90, in <genexpr>
    for name in self._filter(
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/filters.py", line 120, in _filter
    return list(self._check_flows(names))
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/filters.py", line 131, in _check_flows
    check = flow_analysis.reachability_check(
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/flow_analysis.py", line 78, in reachability_check
    return _break_check(context, context_scope, first_flow_scope, node)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/flow_analysis.py", line 93, in _break_check
    reachable = _check_if(context, flow_node)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/flow_analysis.py", line 113, in _check_if
    types = context.eval_node(node)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/base_context.py", line 81, in eval_node
    return self.evaluator.eval_element(self, node)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/__init__.py", line 233, in eval_element
    return self._eval_element_if_evaluated(context, element)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/__init__.py", line 245, in _eval_element_if_evaluated
    return self._eval_element_cached(context, element)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/cache.py", line 42, in wrapper
    rv = function(obj, *args, **kwargs)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/__init__.py", line 249, in _eval_element_cached
    return eval_node(context, element)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/debug.py", line 82, in wrapper
    return func(*args, **kwargs)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/syntax_tree.py", line 50, in wrapper
    return func(context, *args, **kwargs)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/syntax_tree.py", line 148, in eval_node
    return eval_or_test(context, element)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/syntax_tree.py", line 342, in eval_or_test
    types = context.eval_node(next(iterator))
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/base_context.py", line 81, in eval_node
    return self.evaluator.eval_element(self, node)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/__init__.py", line 233, in eval_element
    return self._eval_element_if_evaluated(context, element)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/__init__.py", line 245, in _eval_element_if_evaluated
    return self._eval_element_cached(context, element)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/cache.py", line 42, in wrapper
    rv = function(obj, *args, **kwargs)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/__init__.py", line 249, in _eval_element_cached
    return eval_node(context, element)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/debug.py", line 82, in wrapper
    return func(*args, **kwargs)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/syntax_tree.py", line 50, in wrapper
    return func(context, *args, **kwargs)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/syntax_tree.py", line 148, in eval_node
    return eval_or_test(context, element)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/syntax_tree.py", line 359, in eval_or_test
    types = _eval_comparison(context.evaluator, context, types, operator,
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/syntax_tree.py", line 410, in _eval_comparison
    return ContextSet.from_sets(
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/common/context.py", line 32, in from_sets
    for set_ in sets:
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/syntax_tree.py", line 411, in <genexpr>
    _eval_comparison_part(evaluator, context, left, operator, right)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/syntax_tree.py", line 463, in _eval_comparison_part
    l_is_num = is_number(left)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/helpers.py", line 224, in is_number
    return _get_safe_value_or_none(context, (int, float)) is not None
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/helpers.py", line 210, in _get_safe_value_or_none
    value = context.get_safe_value(default=None)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/base_context.py", line 245, in __getattr__
    return getattr(self._wrapped_context, name)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/utils.py", line 69, in wrapper
    return func(*args, **kwds)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/cache.py", line 143, in wrapper
    result = method(self, *args, **kwargs)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/base_context.py", line 253, in _wrapped_context
    return self._get_wrapped_context()
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/context/iterable.py", line 211, in _get_wrapped_context
    c, = GenericClass(klass, self._get_generics()).execute_annotation()
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/context/iterable.py", line 206, in _get_generics
    return (self.merge_types_of_iterate().py__class__(),)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/base_context.py", line 53, in merge_types_of_iterate
    return ContextSet.from_sets(
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/common/context.py", line 32, in from_sets
    for set_ in sets:
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/base_context.py", line 53, in <genexpr>
    return ContextSet.from_sets(
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/context/iterable.py", line 382, in py__iter__
    for addition in check_array_additions(self._defining_context, self):
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/context/iterable.py", line 640, in check_array_additions
    return _check_array_additions(context, sequence)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/cache.py", line 42, in wrapper
    rv = function(obj, *args, **kwargs)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/debug.py", line 82, in wrapper
    return func(*args, **kwargs)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/context/iterable.py", line 707, in _check_array_additions
    found = evaluate_call_of_leaf(
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/helpers.py", line 102, in evaluate_call_of_leaf
    values = context.eval_node(base)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/base_context.py", line 81, in eval_node
    return self.evaluator.eval_element(self, node)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/__init__.py", line 233, in eval_element
    return self._eval_element_if_evaluated(context, element)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/__init__.py", line 245, in _eval_element_if_evaluated
    return self._eval_element_cached(context, element)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/cache.py", line 42, in wrapper
    rv = function(obj, *args, **kwargs)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/__init__.py", line 249, in _eval_element_cached
    return eval_node(context, element)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/debug.py", line 82, in wrapper
    return func(*args, **kwargs)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/syntax_tree.py", line 50, in wrapper
    return func(context, *args, **kwargs)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/syntax_tree.py", line 74, in eval_node
    return eval_atom(context, element)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/syntax_tree.py", line 200, in eval_atom
    return context.py__getattribute__(
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/base_context.py", line 72, in py__getattribute__
    return f.find(filters, attribute_lookup=not search_global)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/finder.py", line 67, in find
    types = self._names_to_types(names, attribute_lookup)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/finder.py", line 190, in _names_to_types
    contexts = ContextSet.from_sets(name.infer() for name in names)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/common/context.py", line 32, in from_sets
    for set_ in sets:
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/finder.py", line 190, in <genexpr>
    contexts = ContextSet.from_sets(name.infer() for name in names)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/names.py", line 143, in infer
    return tree_name_to_contexts(parent.evaluator, parent, self.tree_name)
  File "/home/werner/.pyenv/versions/miniconda3-latest/envs/spyder.git/lib/python3.8/site-packages/jedi/evaluate/syntax_tree.py", line 622, in tree_name_to_contexts
    raise ValueError("Should not happen. type: %s" % typ)

ValueError: Should not happen. type: del_stmt
```

**Comment by ccordoba12** [comment]
Please install parso version **0.5.2** to fix this problem.

**Comment by hongyi-zhao** [comment]
Thanks, it does the trick.

But, how can I know which version of the dependencies should I use in general? 

Regards

**Comment by ccordoba12** [comment]
This is a mismatch that we noticed a week ago and it'll be fixed in the next conda build of 4.0.1. Pip users will have to wait until 4.1.0 is released though.
